TASK-4/Program.cs

```diff
@@ -0,0 +1,43 @@
+﻿using Persons;
+
+var random = new Random();
+
+var numberOfObjects = random.Next(5, 20);
+
+var listOfObjectsCandidates = new List<Candidate>();
+var listOfObjectsEmployees = new List<Employee>();
+for (var i = 0; i < numberOfObjects; i++)
+{
+    var candidate = new Candidate();
+    
+    listOfObjectsCandidates.Add(candidate);
+    var employee = new Employee();
+    
+    listOfObjectsEmployees.Add(employee);
+}
+
+// заполненение первичных данных из библиотеки
+var repository = new PersonsRepository();
+
+var candidates = repository.GetCandidates(listOfObjectsCandidates);
+
+var employees = repository.GetEmployees(listOfObjectsEmployees);
+
+Console.WriteLine("\n");
+
+candidates.ElementAt(0).ShowUserInfo();
+
+Console.WriteLine("\n");
+
+var reportEmployees = new EmployeeReportGenerator(employees);
+reportEmployees.Report();
+
+Console.WriteLine("\n");
+var reportCandidates = new CandidateReportGenerator(candidates);
+
+reportCandidates.Report();
+
+Console.WriteLine("\n");
+var userFactory = new UserFactory();
+
+userFactory.UserObject();
```

Comment:
Acc to the task and to the factory, you need to return in output employee or candidate according to the person type

Reply:
Unfortunately I really don't have a single idea on how to do this. Can you give me a hint please?

Reply:
In UserFactory must be a method that, depending on the type, will return generated type of user(Employes or Candidates)
These lines should be removed.

Reply:
Removed the lines. Tried to create such a method.